streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamSessionWindowAggregate.java

```diff
@@ -122,11 +124,11 @@ public void process(final K key, final V value) {
             if (!mergedWindow.equals(newSessionWindow)) {
                 for (final KeyValue<Windowed<K>, Agg> session : merged) {
                     store.remove(session.key);
-                    tupleForwarder.maybeForward(session.key, null, session.value);
+                    tupleForwarder.maybeForward(session.key, null, session.value, context().timestamp()); // TODO set proper result timestamp
```

Comment:
If not `context().timestamp()` where would we get the correct timestamp for comparison/calculation?

Reply:
ah, yeah thanks for clearing up, kind of obvious in hindsight